server/src/main/java/org/elasticsearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -86,6 +86,8 @@ public class PublicationTransportHandler {
         TransportRequestOptions.Type.STATE
     );
 
+    public static final Version INCLUDES_LAST_COMMITTED_DATA_VERSION = Version.V_8_7_0;
```

Comment:
@DaveCTurner should this be 8.6.0 since it's intended to be backported?

Reply:
I'll adjust that while backporting. If I set this to 8.6.0 now then the BwC test suite would fail on this PR, but it's important for those tests to pass because they tell us whether there are any other more subtle BwC problems here.

Reply:
Joining the dots here:

- I opened a PR at https://github.com/elastic/elasticsearch/pull/92277 which backports this to 8.6 with a fixed BwC version, and waited for it to pass CI.
- I merged 296eacd1a3b2c41275dddafc3469f60d309d7344 which disables the BwC tests in `main`.
- I merged the backport PR.
- I opened another PR at https://github.com/elastic/elasticsearch/pull/92280 to fix up the BwC version in `main` and reinstate the BwC tests, which will be merged once it passes CI.